qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -119,7 +125,73 @@ protected static Distribution distribution() {
         return distribution;
     }
 
-    protected Shell newShell() throws Exception {
+    protected static void install() throws Exception {
+        switch (distribution.packaging) {
+            case TAR:
+            case ZIP:
+                installation = Archives.installArchive(distribution);
+                Archives.verifyArchiveInstallation(installation, distribution);
+                break;
+            case DEB:
+            case RPM:
+                installation = Packages.installPackage(distribution);
+                Packages.verifyPackageInstallation(installation, distribution, newShell());
+                break;
+            case DOCKER:
+                installation = Docker.runContainer(distribution);
+                Docker.verifyContainerInstallation(installation, distribution);
+        }
+    }
+
+    protected Shell.Result runTool(String tool, String args) throws Exception {
+        if (Platforms.WINDOWS) {
+            tool += ".bat";
+        }
+        Path path = installation.bin(tool);
+        return sh.run(path.toString() + " " + args);
+    }
+
+    protected void assertWhileRunning(Platforms.PlatformAction assertions) throws Exception {
```

Comment:
I think it would be helpful to have some javadocs on this method indicating the method runs elasticsearch, performs the assertions, and then stops elasticsearch